pkg/operator/controller/externaldns/pod.go

```diff
@@ -139,12 +139,8 @@ func (b *externalDNSContainerBuilder) fillProviderAgnosticFields(seq int, zone s
 		args = append(args, fmt.Sprintf("--namespace=%s", *b.externalDNS.Spec.Source.Namespace))
 	}
 
-	if len(b.externalDNS.Spec.Source.AnnotationFilter) > 0 {
-		annotationFilter := ""
-		for key, value := range b.externalDNS.Spec.Source.AnnotationFilter {
-			annotationFilter += fmt.Sprintf("%s=%s,", key, value)
-		}
-		args = append(args, fmt.Sprintf("--annotation-filter=%s", annotationFilter[0:len(annotationFilter)-1]))
+	if b.externalDNS.Spec.Source.LabelFilter != nil {
+		args = append(args, fmt.Sprintf("--label-filter=%s", metav1.FormatLabelSelector(b.externalDNS.Spec.Source.LabelFilter)))
```

Comment:
Actually no, since they will be array values in the manifest adding a single quote into the string means it will be read by the controller as a value with quotes. These arguments are not processed by a shell before being passed to the command in the image.